repoparse.py

```diff
@@ -0,0 +1,294 @@
+            try:
+                diff_str += unwrap_bytes_gen_to_str(
+                    unified_diff(
+                        s_get_lines(old_content),
+                        s_get_lines(new_content),
+                        p_old_path,
+                        p_new_path
+                    )
+                )
+            except UnicodeDecodeError:
+                continue
+
+    parsed_commits = parsed_repo[commit.author]
+    if parsed_commits is None:
+        parsed_commits = parsed_repo[commit.author] = ParsedCommits({})
+
+    try:
+        patch_languages = {}
+        patch_set_dict  = {}
+        patch_ids_dict  = {}
+
+        # Splice out the parsed files
+        patch_files = patch_set_to_json(repo.path, PatchSet(diff_str))
+
+        for f in patch_files:
+            file_path = path.join(repo.path, f)
+            if file_path in enry_dict:
+                languages = enry_dict[file_path]
+                if (ll := languages[0].lower()) in lang_objs:
+                    patch_text = patch_files[file_path]
+
+                    parser.set_language(lang_objs[ll])
+                    treesitter_query = lang_objs[ll].query('((identifier) @id)')
+                    patch_string = cleanup_diff(patch_text)
+
+                    src_bytes = bytes(patch_string, 'utf8')
+                    captures = treesitter_query.captures(parser.parse(src_bytes).root_node)
+                    if captures:
+                        patch_set_dict[file_path] = patch_string
+                        patch_languages[file_path] = ll
+                        patch_ids_dict[file_path] = list(map(lambda capture: src_bytes[capture[0].start_byte:capture[0].end_byte].decode('utf8'), captures))
+
+        if patch_languages and patch_set_dict and patch_ids_dict:
+            parsed_commits[commit.sha().hexdigest()] = ParsedCommitInfo(
+                parent.sha().hexdigest(),
+                patch_set_dict,
+                patch_languages,
+                patch_ids_dict
+            )
+
+    except UnidiffParseError as e:
+        with open(f'out/error{chunk_n}:{curr_item}.log', 'w') as f:
+            f.write(diff_str)
+        sys.stderr.write(f'\n!!! ERROR !!!: Invalid (?) unidiff at item {chunk_n}:{curr_item}, hex: {commit.sha().hexdigest()}\n')
+        sys.stderr.write(f'\nText:\n\t{e}\nSee error{chunk_n}:{curr_item}.log for details...\n')
+
+    if not quiet:
+        progress(commit.sha().hexdigest(), curr_item, 'unk')
+    
+    return True
+
+def parse_repo_commits(repo_path: str, enry_dict: dict, quiet: bool, chunked: bool = False, chunk_size: int = 10000):
+    """Parse a repo.
+    :repo_path: A path to the repo on the current os's FS.
+    """
+    if not chunked:
+        return _parse_repo_commits_unchunked(repo_path, enry_dict, quiet)
+    else:
+        return _parse_repo_commits_chunked(repo_path, enry_dict, quiet, chunk_size)
+
+def clone_and_detect_path(repo_path, force_reclone):
+
+    if not path.exists('build'):
```

Comment:
you can store everything you need to work in module (`.so` and so on) - but everything else should be stored in other place - and use an argument to specify location to store repositories and other pipeline's artifacts